aptos-move/framework/aptos-stdlib/sources/cryptography/bulletproofs.move

```diff
@@ -0,0 +1,266 @@
+/// This module implements Bulletproof-based zero-knowledge range proof: i.e., a proof that a value `v` committed in a
+/// Pedersen commitment `com` satisfies $v \in [0, 2^{num_bits})$. Currently, this module only supports num_bits \in
+/// {8, 16, 32, 64}.
+module aptos_std::bulletproofs {
+    use std::features;
+    use aptos_std::elgamal;
+    use aptos_std::pedersen;
+    use aptos_std::ristretto255::{Self, RistrettoPoint};
+
+    //
+    // Constants
+    //
+
+    /// The maximum range supported by the Bulletproofs library is [0, 2^{64}).
+    const MAX_RANGE_BITS : u64 = 64;
+
+    //
+    // Error codes
+    //
+
+    /// Error deserializing one of the arguments.
+    const E_DESERIALIZE_RANGE_PROOF: u64 = 1;
+
+    /// The input value given to the prover is too large.
+    const E_VALUE_OUTSIDE_RANGE: u64 = 2;
+
+    /// The range proof system only supports proving ranges of type [0, 2^{bits}) where bits \in {8, 16, 32, 64}.
+    const E_RANGE_NOT_SUPPORTED: u64 = 3;
+
+    /// The native functions have not been rolled out yet.
+    const E_NATIVE_FUN_NOT_AVAILABLE: u64 = 4;
+
+    //
+    // Structs
+    //
+
+    /// Represents a zero-knowledge range proof that a value committed inside a Pedersen commitment lies in [0, 2^{MAX_RANGE_BITS}).
+    struct RangeProof has copy, drop, store {
+        bytes: vector<u8>
+    }
+
+    //
+    // Public functions
+    //
+
+    /// Returns the maximum # of bits the range proof system can verify proofs for.
+    public fun get_max_range_bits(): u64 {
+        MAX_RANGE_BITS
+    }
+
+    /// Deserializes a range proof from a sequence of bytes.
+    public fun range_proof_from_bytes(bytes: vector<u8>): RangeProof {
+        RangeProof {
+            bytes
+        }
+    }
+
+    /// Returns the byte-representation of a range proof.
+    public fun range_proof_to_bytes(proof: &RangeProof): vector<u8> {
+        proof.bytes
+    }
+
+    /// Verifies a zero-knowledge range proof that the value `v` committed in `com` (under the default Bulletproofs
+    /// commitment key; see `pedersen::new_commitment_for_bulletproof`) satisfies $v \in [0, 2^{num_bits})$. Only works
+    /// for `num_bits` \in {8, 16, 32, 64}.
+    public fun verify_range_proof_pedersen(com: &pedersen::Commitment, proof: &RangeProof, num_bits: u64, dst: vector<u8>): bool {
+        if(!features::bulletproofs_enabled()) {
+            abort(std::error::invalid_state(E_NATIVE_FUN_NOT_AVAILABLE))
+        };
+
+        verify_range_proof_internal(
+            ristretto255::point_to_bytes(&pedersen::commitment_as_compressed_point(com)),
+            &ristretto255::basepoint(), &ristretto255::hash_to_point_base(),
```

Comment:
I propose a small gas optimization here: actually define the serialized point in a constant in this module, and have this function simply deserialize it. In the comments, you can explain that this point is obtained by doing `        new_point_from_sha2_512(basepoint_compressed().data)`.

This would avoid the extra gas cost associated with the hashing.